tests/RobotFramework/tests/cumulocity/software.robot

```diff
@@ -4,25 +4,26 @@ Library    Cumulocity
 Library    ThinEdgeIO
 
 Test Tags    theme:c8y    theme:software    theme:plugins
+Test Setup       Custom Setup
 Test Teardown    Get Logs
 
 *** Test Cases ***
 Software list should be populated during startup
-    ${DEVICE_SN}=                            Setup
-    Device Should Exist                      ${DEVICE_SN}
-    Device Should Have Installed Software    tedge
+    Device Should Have Installed Software    tedge    timeout=120
 
 Install software via Cumulocity
-    ${DEVICE_SN}=                            Setup
-    Device Should Exist                      ${DEVICE_SN}
-    Sleep    12s    reason=FIX: Wait for device to be ready for operations
     ${OPERATION}=    Install Software        c8y-remote-access-plugin    # TODO: Use different package
     Operation Should Be SUCCESSFUL           ${OPERATION}    timeout=60
     Device Should Have Installed Software    c8y-remote-access-plugin
 
 Software list should only show currently installed software and not candidates
-    [Tags]    flakey
+    ${EXPECTED_VERSION}=    Execute Command    dpkg -s tedge | grep "^Version: " | cut -d' ' -f2    strip=True
+    Device Should Have Installed Software    tedge,^${EXPECTED_VERSION}::apt$        timeout=120
+
+*** Keywords ***
+
+Custom Setup
     ${DEVICE_SN}=                            Setup
     Device Should Exist                      ${DEVICE_SN}
-    ${EXPECTED_VERSION}=    Execute Command    dpkg -s tedge | grep "^Version: " | cut -d' ' -f2    strip=True
-    Device Should Have Installed Software    tedge,^${EXPECTED_VERSION}::apt$
+    Set Test Variable    $DEVICE_SN
+    Should Have MQTT Messages    tedge/health/tedge-mapper-c8y
```

Comment:
This makes test more deterministic than just a wait.

Reply:
That is the idea, however it seems to be flakey as the software list req message sometimes goes unanswered by the `tedge-agent` service. I'm trying to find another way (and will raise a bug ticket)